code/client/src/pages/Show/Swap.jsx

```diff
@@ -52,27 +130,60 @@ const getSelectedTokenComputedBalance = (selectedToken, tokenBalances, oneWallet
   }
 }
 
-const textStyle = {
-  paddingRight: '10px',
-  display: 'block'
+/**
+ * Handles a swap amount change, it can be either swap amount or target swap amount.
+ * A swap amount or target swap amount is automatically set based on the provided exchange rate.
+ */
+const handleSwapAmountChange = ({
+  value,
+  exchangeRate,
+  setSwapAmountFunc,
+  setCalculatedAmountFunc
+}) => {
+  if (isNaN(value)) {
+    setSwapAmountFunc(undefined)
+  } else {
+    const amount = util.formatNumber(value, 4)
+
+    setSwapAmountFunc(value)
+
+    const calculatedAmount = util.formatNumber(amount * exchangeRate, 4)
+
+    if (exchangeRate && !isNaN(calculatedAmount)) {
+      setCalculatedAmountFunc(calculatedAmount.toString())
+    }
+  }
 }
 
-const optionButtonStyle = {
-  textAlign: 'left',
-  height: '50px'
-}
+/**
+ * Handles swap token selection. Either a "swap from" token or a "swap to" token can be handled.
+ * TODO: actual exchange rate should be fetched and updated here.
+ * If the selected token changes, the target swap amount will be automatically updated based on fetched exchange rate.
+ */
+const handleSwapTokenSelected = ({
```

Comment:
If this function and the one above depend on so many state variables and functions, it might make more sense to put them inside the component. The re-rendering should cause minimal impact on performance.